### PR TITLE
Mobile menu fix

### PR DESCRIPTION
**Pull request by noahmanger**
## Summary

This fixes a few problems under the hood with the mobile menu:
- Sets position relative to the header, rather than the body, which is better for eregs
- Fixes a bug where utility nav items and the menu title weren't showing up if you started at a big display and then shrank it

In order to fix that second one, I forked the mega menu library and [edited it slightly](https://github.com/noahmanger/Accessible-Mega-Menu/commit/969e51be809bb6a4a481befc0c31ca8089c2c945) to allow for a selector to be passed through to designate which nav items would actually contain mega menus. This solves the problem where the mega menu expected every `li` to contain an item, so was appending various ARIA attributes that they shouldn't have (like `aria-hidden`), which was causing them to not show up. This has the added benefit of not including the menu library in this repo.

I decided to go with a fork rather than suggesting an upstream fix because they don't seem to be maintaining that repo anymore.

Resolves https://github.com/18F/fec-style/issues/391
Resolves https://github.com/18F/fec-style/issues/390

cc @adborden

### CODE REVIEW

File: tests/site-nav.js [hunk 1]
```diff
@@ -46,11 +46,11 @@ describe('SiteNav', function() {
 
   describe('constructor()', function() {
     it('should set body to jqueryized selector', function() {
-      expect(this.siteNav.$body).to.be.ok;
-      expect(this.siteNav.$body.length).to.be.ok;
```
Comment (adborden) [inline tests/site-nav.js]: These tests are still valid, right? (we still have a `$body` property).

File: js/site-nav.js [hunk 2]
```diff
@@ -113,19 +117,21 @@ SiteNav.prototype.toggleMenu = function() {
...
+  this.$body.css({'overflow': 'hidden'});
+  this.$element.addClass('is-open');
   this.$toggle.addClass('active');
   this.$menu.attr('aria-hidden', false);
   this.isOpen = true;
 };
 
 SiteNav.prototype.hideMenu = function() {
-  this.$body.removeClass('is-open');
+  this.$body.css({'overflow': 'auto'});
```
Comment (adborden) [inline js/site-nav.js]: Would be good to add tests that check the `overflow` is toggled.

**Comment by adborden** [comment]
:+1: What's the plan for the mega menu? Do you think you/18F should take ownership of it? Would be nice to get it into npm proper. Maybe you should open an issue in their repo asking if they are going to continue support for it?

File: tests/site-nav.js (hunk 1 above)
Comment (noahmanger) [inline tests/site-nav.js]: Yeah, I just renamed it to `$element`. The test stayed the same.

**Comment by noahmanger** [comment]
Good question on the mega menu. I'm not sure what's best. I'll ask in Slack.

File: js/site-nav.js (hunk 2 above)
Comment (noahmanger) [inline js/site-nav.js]: 👍

File: tests/site-nav.js (hunk 1 above)
Comment (adborden) [inline tests/site-nav.js]: Oh, I mean that these lines could stay as is, having the same test for `siteNav.$body` _and_ for `siteNav.$element`.
